src/libraries/Microsoft.Win32.SystemEvents/src/Microsoft/Win32/LocalAppContextSwitches.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System;
+
+internal static partial class LocalAppContextSwitches
+{
+    private static int s_enableLegacySystemEventsShutdownThreadJoin;
+
+    public static bool EnableLegacySystemEventsShutdownThreadJoin
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get => GetCachedSwitchValue("Switch.SystemEvents.EnableLegacySystemEventsShutdownThreadJoin", ref s_enableLegacySystemEventsShutdownThreadJoin);
```

Comment:
Do you expect that this fix is likely going to break applications out there?

If we are going to introduce a config switch for this, are we also going to file a breaking change notice for this that explains when people should use this config switch?

Reply:
While I think it is unlikely, I suppose that there could be an app out there that absolutely needs to know when the SystemEvents thread has shutdown (though given how the code was in Framework, there wouldn't be a good way to know that for STA threads so this makes chances even slimmer). I'm not sure what scenario that would be and it would be nice to know about the specifics of the scenario if such cases exist. We could introduce the switch later if we get a report. That way we'd also have better understanding of the scenario.

Reply:
Yes, I think introducing config switches reactively only once there is actual compat problem is a better strategy. It keeps the code simple.

Reply:
I'm OK without the compat switch for 10.0 - but if we backport to servicing I think we might need it.

We will have a breaking change with this - we discussed that we can no longer guarantee that `EventsThreadShutdown` will be raised.